ledger/src/shred/merkle.rs

```diff
@@ -956,6 +961,9 @@ pub(super) fn recover(
         .map(Shred::merkle_node)
         .collect::<Result<_, _>>()?;
     let tree = make_merkle_tree(nodes);
+    // The attched signature verfies only if we obtain the same Merkle root.
+    // Because shreds obtained from turbine or repair are sig-verified, this
+    // also means that we don't need to verify signatures for recovered shreds.
     if tree.last() != Some(&merkle_root) {
         return Err(Error::InvalidMerkleRoot);
```

Comment:
Now that we compare signatures is this clause impossible?
Not saying we should remove it, just want to understand the edge cases

Reply:
That is a good question.
I don't think it is possible but I cannot prove it.